public/app/percona/dbaas/components/DBCluster/DBCluster.tsx

```diff
@@ -133,8 +133,8 @@ export const DBCluster: FC = () => {
 
   const addCluster = async (values: Record<string, any>, showPMMAddressWarning: boolean) => {
     try {
-      await dispatch(addDbClusterAction({ values, setPMMAddress: showPMMAddressWarning })).unwrap();
       setAddModalVisible(false);
+      await dispatch(addDbClusterAction({ values, setPMMAddress: showPMMAddressWarning })).unwrap();
```

Comment:
what if we have an error adding a cluster? Wouldn't users have to re-open the modal and insert all values again?

Reply:
@fabio-silva we can also change it now just for delete option

Reply:
@solovevayaroslavna yeah, I think we can then do that for the deletion. Maybe remove that `thunkAPI.dispatch` from there, and call the fetch after the deletion in the component